src/pineko/scaffold.py

```diff
@@ -0,0 +1,71 @@
+"""Tools related to generation and managing of a pineko project."""
+
+import os
+import pathlib
+
+from .configs import needed_files, needed_keys
+
+
+def set_up_project(configs):
+    """Set up all the folders spelled out in the configs dictionary.
+
+    Parameters
+    ----------
+    configs : dict
+        configs dictionary containing all the paths to be set up
+    """
+    for path in configs["paths"]:
+        if path == "root" or path in needed_files:
+            continue
+        if isinstance(configs["paths"][path], pathlib.Path):
+            configs["paths"][path].mkdir(parents=True, exist_ok=True)
+        elif isinstance(configs["paths"][path], dict):
+            for log_path in configs["paths"][path]:
+                if isinstance(configs["paths"][path][log_path], pathlib.Path):
+                    os.makedirs(configs["paths"][path][log_path], exist_ok=True)
+                else:
+                    raise TypeError(f"Not recognized entry {log_path} in configs")
+        else:
+            raise TypeError(f"Not recognized entry {path} in configs")
+
+
+def check_folders(configs):
+    """Check if all the folders spelled out in configs exist.
+
+    Parameters
+    ----------
+    configs : dict
+        configs dictionary containing all the paths to be checked
+    Returns
+    -------
+    : bool
+        True if the configuration is correct, False otherwise
+    : list
+        list of missing keys in the configuration file
+    : dict
+        dictionary containing folders that should exists but could
+        not be found
```

Comment:
It is simpler to write a very brief dataclass, let's say `CheckResult`, and return this one. Something like:
```py
@dataclasses.dataclass
class CheckResult:
    confs: List[pathlib.Path]
    folders: Dict[str, pathlib.Path]
```